skyportal/handlers/api/tns.py

```diff
@@ -254,6 +263,25 @@ def tns_retrieval(obj_id, tnsrobot_id, user_id, include_spectra=False):
             source_data = r.json().get("data", dict()).get("reply", dict())
             if source_data:
                 obj.tns_info = source_data
+                group_ids = [g.id for g in user.accessible_groups]
+
+                if include_photometry and 'photometry' in source_data:
+                    photometry = source_data['photometry']
+                    for phot in photometry:
+                        try:
+                            df, instrument_id = read_tns_photometry(phot, session)
+                            data_out = {
+                                'obj_id': obj_id,
+                                'instrument_id': instrument_id,
+                                'group_ids': group_ids,
+                                **df.to_dict(orient='list'),
+                            }
+                            add_external_photometry(data_out, user)
+                        except Exception as e:
+                            log(
+                                f'Cannot read TNS photometry {str(photometry)}: {str(e)}'
```

Comment:
Here I'd use flow to send a frontend messages. You can maybe store failed ingestions info in a list to concatenate into one error message you can send back to the frontend after iterating. That way we can display a message on the frontend (well of course, if this was triggered from the frontend). This is to avoid us from having to check the logs manually every time someone doesn't get the data they expected to ingest.